### PR TITLE
Fix the Tutorial4 plots macro

### PR DESCRIPTION
**Pull request by karabowi**
Bug fixing.

Addresses issue #1209.

---

Checklist:

[ X ] Rebased against `dev` branch
[ X ] My name is in the resp. CONTRIBUTORS/AUTHORS file
[ X ] Followed [the seven rules of great commit messages](https://chris.beams.io/posts/git-commit/#seven-rules)

### CODE REVIEW

File: examples/simulation/Tutorial4/macros/plots.C [hunk 1]
```diff
@@ -29,8 +31,8 @@ plots(Int_t nEvents = 1000, Int_t iout = 1, TString mcEngine = "TGeant3")
     TFile *f1 = TFile::Open(MCFile);
     TFile *f2 = TFile::Open(RecoFile);
 
-    TTree *t1 = f1->Get("cbmsim");
-    TTree *t2 = f2->Get("cbmsim");
+    TTree *t1 = (TTree *)f1->Get("cbmsim");
```
Comment (fuhlig1) [inline examples/simulation/Tutorial4/macros/plots.C:34]: Please use C++ casts and not the C one. 
Or even better use the templated Get function

TTree* t1    = f1->Get<TTree>("cbmsim");

File: examples/simulation/Tutorial4/macros/plots.C [hunk 2]
```diff
@@ -132,11 +134,9 @@ plots(Int_t nEvents = 1000, Int_t iout = 1, TString mcEngine = "TGeant3")
         fHist->Write();
         if (0) {   // explicit writing
             TIter next(gDirectory->GetList());
-            TH1 *h;
-            TObject *obj;
-            while (obj = (TObject *)next()) {
+            while (TObject *obj = (TObject *)next()) {
                 if (obj->InheritsFrom(TH1::Class())) {
-                    h = (TH1 *)obj;
+                    TH1 *h = (TH1 *)obj;
```
Comment (fuhlig1) [inline examples/simulation/Tutorial4/macros/plots.C:139]: C++ cast.

**Comment by ChristianTackeGSI** [comment]
Can we try to run this macro in the testsuite?

**Comment by karabowi** [comment]
I will make the changes and add the macro to the test suite.

File: examples/simulation/Tutorial4/macros/plots.C (hunk 1 above)
Comment (ChristianTackeGSI) [reply examples/simulation/Tutorial4/macros/plots.C:34]: Maybe?

```c++
auto t1 = f1->Get<TTree>("cbmsim");
```

Or maybe
```c++
std::unique_ptr<TTree> t1{f1->Get<TTree>("cbmsim")};
```

**Review by fuhlig1 — APPROVED** [review]
After removing the C style casts I am fine with the MR

**Comment by karabowi** [comment]
Removed the piece of unused code and changed the return value.

**Pull request merged**